File: C plus/kickstartC1.cpp/p2.py
def partition(N, PartitionSum):
    assert(N >= 0 and PartitionSum >= 0)
    if (PartitionSum == 0 or N == 0):
      return []
    if(N > PartitionSum):
      return partition(N-1, PartitionSum)
    else:
      return [N] + partition(N-1, PartitionSum-N)
t = int(input())
for a in range(1,t+1):
    n,x,y = [int(b) for b in input().split()]
    sum1 = n*(n+1)
    sum1 = sum1//2
    if(sum1%(x+y) == 0):
        suma = (x)*(sum1)//(x+y)
        seq = partition(n,suma)
        result = ""
        for num in seq:
            result += str(num)+" "
        print("Case #"+str(a)+": POSSIBLE")
        print(len(seq))
        print(result)
    else:
        print("Case #"+str(a)+": IMPOSSIBLE")

File: C plus/kickstartC1.cpp/p2.cpp
#include<bits/stdc++.h>
using namespace std;
int t,n,x,y;

vector<int> p(int n,int sum){
    if(n == 0 || sum == 0){
        vector<int> s = {};
        return s;
    }
    if(n > sum){
        return p(n-1,sum);
    }else{
        vector<int> t = p(n-1,sum-n);
        t.push_back(n);
        return t;
    }
}
int main(){
    cin >> t;
    for(int a = 1;a<t+1;a++){
        cin >> n >> x >> y;
        int sum = (n)*(n+1)/2;
        if(sum%(x+y) == 0){
            int suma = (x)/(x+y)*sum;
            vector<int> result = p(n,suma);
            string results = "";
            for(int i = 0;i<result.size();i++){
                results += result[i]+" ";
            }
            cout << "Case #"<< a << ": " << "POSSIBLE" << endl; 
            cout << result.size() << endl;
            cout << results << endl;
        }else{
           cout << "Case #"<< a << ": " << "IMPOSSIBLE" << endl ; 
        }

    }
    return 0;
}